MeretzLib/Apple/iOS/MeretzSampleGame/MeretzSampleGame/GameViewController.m

```diff
@@ -8,351 +8,766 @@
+	// any saved access token is now invalid.
+	- (void) didVendorUserDisconnectFinish:(MeretzResult *)result;
+	// called when a vendorConsumeWithinRange task finishes
+	// if successful (result.Success==TRUE), result.Items contains all owned items within specified date range.
+	- (void) didVendorConsumeWithinRangeFinish:(MeretzVendorConsumeResult *)result;
+	// called when a vendorConsumeGetNew task finishes
+	// if successful (result.Success==TRUE), result.Items contains any newly acquired (un-consumed) items.
+	-(void) didVendorConsumeGetNewFinish:(MeretzVendorConsumeResult *)result;
+	// called when a vendorConsumeAcknowledge task finishes
+	// if successful (result.Success==TRUE), result.Items contains the acknowledged (consumed) items.
+	-(void) didVendorConsumeAcknowledgeFinish:(MeretzVendorConsumeResult *)result;
+
+@end
+
 @implementation GameViewController
 
-Meretz *gMeretz= nil;
-MeretzSampleGameState gSampleGameState= Initial;
-MeretzTaskId gConnectUserTaskId= MERETZ_TASK_ID_INVALID;
-MeretzTaskId gVendorConsumeTaskId= MERETZ_TASK_ID_INVALID;
-MeretzTaskId gVendorDisconnectTaskId= MERETZ_TASK_ID_INVALID;
+	@synthesize MeretzAPI;
+	@synthesize SampleGameState;
+	@synthesize VendorConnectUserTaskId;
+	@synthesize VendorConsumeTaskId;
+	@synthesize VendorDisconnectUserTaskId;
+	@synthesize ConnectionCodeTextField;
+	@synthesize ConnectionCode;
 
-static void updateSamepleGameState(void)
-{
-	switch (gSampleGameState)
+	-(void) initializeMeretzDemo
 	{
-		case Initial:
+		// if we have previously saved a Meretz Access Token, use it
+		// your game will want to use a similar pattern
+		NSString *savedAccessToken= [self getSavedMeretzAccessToken];
+		if ((nil != savedAccessToken) && (0 < [savedAccessToken length]))
 		{
-			// initiate a user connection task
-			NSString *userConnectionCode= TEST_MERETZ_USER_CONNECTION_CODE;
-			NSString *vendorUserIdentifier= SAMPLE_GAME_USER_IDENTIFIER;
+			NSLog(@"Retrieved saved Meretz access token: %@", savedAccessToken);
+		}
+		
+		id useDelegate= USE_MERETZ_DELEGATE ? self : nil;
+		
+		// initialize Meretz lib
+		NSLog(@"Initializing Meretz");
+		self.MeretzAPI= [[Meretz alloc] init:savedAccessToken optionalDelegate:useDelegate];
+		NSAssert(nil != self.MeretzAPI, @"Failed to initialize MeretzLib!");
+		
+		// setup some demo app state
+		self.SampleGameState= Initial;
+		self.VendorConnectUserTaskId= MERETZ_TASK_ID_INVALID;
+		self.VendorConsumeTaskId= MERETZ_TASK_ID_INVALID;
+		self.VendorDisconnectUserTaskId= MERETZ_TASK_ID_INVALID;
+		
+		self.ConnectionCodeTextField= nil;
+		self.ConnectionCode= @"";
+		
+		if (FALSE)
+		{
+			// point Meretz at a custom dev server
+			[MeretzAPI setMeretzHostName:@"127.0.0.1"];
+			[MeretzAPI setMeretzPort:8000];
+			[MeretzAPI setMeretzProtocol: @"http"];
+			[MeretzAPI setMeretzAPIPath:@""];
+			NSLog(@"Meretz server set to: %@", [MeretzAPI getMeretzServerString]);
+		}
+		
+		[self initializeMeretzDemoUI];
+		
```

Comment:
Sample now runs a simple update function in a timer-driven loop, for periodic state checking